src/docfx/lib/markdown/MarkdownUtility.cs

```diff
@@ -27,28 +29,27 @@ internal static class MarkdownUtility
     {
         private static readonly ConcurrentDictionary<string, Lazy<IReadOnlyDictionary<string, string>>> s_markdownTokens = new ConcurrentDictionary<string, Lazy<IReadOnlyDictionary<string, string>>>(StringComparer.OrdinalIgnoreCase);
 
-        private static readonly Dictionary<MarkdownPipelineType, MarkdownPipeline> s_pipelineMapping =
-            new Dictionary<MarkdownPipelineType, MarkdownPipeline>()
-                {
-                    { MarkdownPipelineType.ConceptualMarkdown, CreateConceptualMarkdownPipeline() },
-                    { MarkdownPipelineType.InlineMarkdown, CreateInlineMarkdownPipeline() },
-                    { MarkdownPipelineType.TocMarkdown, CreateTocPipeline() },
-                    { MarkdownPipelineType.Markdown, CreateMarkdownPipeline() },
-                };
+        private static readonly MarkdownPipeline[] s_markdownPipelines = new[]
+        {
+            CreateConceptualMarkdownPipeline(),
+            CreateInlineMarkdownPipeline(),
+            CreateTocMarkdownPipeline(),
+            CreateMarkdownPipeline(),
+        };
 
         private static readonly ThreadLocal<Stack<Status>> t_status = new ThreadLocal<Stack<Status>>(() => new Stack<Status>());
 
         public static MarkupResult Result => t_status.Value.Peek().Result;
 
-        public static (MarkdownDocument ast, MarkupResult result) Parse(string content)
+        public static (MarkdownDocument ast, MarkupResult result) Parse(string content, MarkdownPipelineType piplineType)
```

Comment:
is that too risk? when someone change the order of pipelineType?

Reply:
we have tests to detect that